### PR TITLE
Filter out non-netdev entries in sysfs/class/net

### LINKED ISSUE #505: v0.9.0 fails to start on Centos6
Running v0.9.0 (latest at docker hub) on Centos6 fails with following error.
```
F0216 04:47:21.812981 00001 cadvisor.go:76] Failed to create a Container Manager: open /sys/class/net/bonding_masters/address: not a directory
goroutine 1 [running]:
github.com/golang/glog.stacks(0x4c2080ae700, 0x0, 0x0, 0x0)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:726 +0xcd
github.com/golang/glog.(*loggingT).output(0xf014a0, 0x4c200000003, 0x4c2080bae40)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:677 +0x24c
github.com/golang/glog.(*loggingT).printf(0xf014a0, 0x4c200000003, 0x9fe4d0, 0x28, 0x4c2080d3f38, 0x1, 0x1)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:635 +0x19c
github.com/golang/glog.Fatalf(0x9fe4d0, 0x28, 0x4c2080d3f38, 0x1, 0x1)
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/Godeps/_workspace/src/github.com/golang/glog/glog.go:1033 +0x64
main.main()
        /usr/local/google/home/vmarmol/go/src/github.com/google/cadvisor/cadvisor.go:76 +0x5fe
```

/sys/class/net/ does contain a file called ```bonding_masters```
```
ls -l /sys/class/net/
...
lrwxrwxrwx 1 root root    0 2015-01-30 10:01 bond0 -> ../../devices/virtual/net/bond0
-rw-r--r-- 1 root root 4096 2015-01-30 10:01 bonding_masters
lrwxrwxrwx 1 root root    0 2015-01-30 10:01 docker0 -> ../../devices/virtual/net/docker0
```

Maybe should skip over non-directory, non-symlink files?

### PR DESCRIPTION
**Pull request by difro**
fixes #505

### CODE REVIEW

File: utils/sysfs/sysfs.go [hunk 1]
```diff
@@ -98,7 +98,25 @@ func (self *realSysFs) GetBlockDeviceSize(name string) (string, error) {
...
+	files, err := ioutil.ReadDir(netDir)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter out non-directory & non-symlink files
+	var dirs []os.FileInfo
+	for _, f := range files {
+		if f.Mode()|os.ModeSymlink != 0 {
+			f, err = os.Stat(netDir + "/" + f.Name())
```
Comment (vmarmol) [inline utils/sysfs/sysfs.go]: nit: path.Join(netDir, f.Name())

**Comment by vmarmol** [comment]
ok to test

**Comment by vmarmol** [comment]
Thanks for the fix @difro! Would you mind signing the [Google CLA](https://github.com/google/cadvisor/blob/master/CONTRIBUTING.md#contributor-license-agreements) before we merge?

**Comment by difro** [comment]
@vmarmol I've submitted the CLA with my github id.

**Comment by vmarmol** [comment]
Awesome, I see it now. Just the one nit and we'll merge.

**Comment by difro** [comment]
@vmarnol thanks. pushed another commit. Ping me if you prefer new forced commit update with both commits merged in.

File: utils/sysfs/sysfs.go [hunk 2]
```diff
@@ -98,7 +98,25 @@ func (self *realSysFs) GetBlockDeviceSize(name string) (string, error) {
...
+
+	// Filter out non-directory & non-symlink files
+	var dirs []os.FileInfo
+	for _, f := range files {
+		if f.Mode()|os.ModeSymlink != 0 {
+			f, err = os.Stat(path.Join(netDir, f.Name()))
+			if err != nil {
+				continue
+			}
+		}
```
Comment (rjnagal) [inline utils/sysfs/sysfs.go]: I don't think this works for a normal case where we just have symlinks:
$ ls -l /sys/class/net/
total 0
lrwxrwxrwx 1 root root 0 Feb 14 18:09 docker0 -> ../../devices/virtual/net/docker0
lrwxrwxrwx 1 root root 0 Jan 20 06:58 eth0 -> ../../devices/pci0000:00/0000:00:04.0/virtio1/net/eth0

eth0 will pass the test for symlink and fail the one for IsDir(). We'll end up not adding it to the known devices.

Comment (difro) [inline utils/sysfs/sysfs.go]: hmm. os.Stat() will follow eth0 symlink and modify f with the symlinked original entry(../../devices/pci0000:00/0000:00:04.0/virtio1/net/eth0) which IS a directory.. no?

Comment (rjnagal) [inline utils/sysfs/sysfs.go]: sorry, I didn't notice that you were modifying 'f'. Getting late :)

This should be fine.

**Comment by vmarmol** [comment]
Please merge the commits. Yeah sorry I'm a bit of a pain :)

**Comment by difro** [comment]
merged & force-pushed.

**Comment by vmarmol** [comment]
LGTM, thanks again @difro!

**Pull request merged**